openfifth/aspen-discovery
# Cookie management db cleanup

This PR clears the user_*_usage tables when the user sets their external search source storage cookies to 0.

TEST PLAN
1. Apply patch
2. Ensure Summon is set up, if it is not:
   1. Admin Administration -> System Administration ->Modules Select Summon, check the 'enabled' box and save. 
   2. In the menu on the left hand side you will now see a section that says Summon. 
   3.. Click on Settings and then Add New. Enter the information for Summon found in Passbolt. 
   4.Use the Admin Menu to navigate to Primary Configuration -> Library Systems. Select your Library, scroll down to Summon and 
  select the name you gave your Summon settings from the dropdown and click save. 
3. Ensure Library Cookie Consent is Required:
   1. Navigate to Primary Configuration -> Library Systems and select your library. 
   2. Scroll down to Data Protection Regulations and ensure the checkbox for Require Cookie Storage Consent is clicked and click
   save. 
4. Navigate to the 'Your Cookie Preferences' section in your account. Turn on 'External Search Services' and save. 
5.  Carry out a Summon Search (the search dropdown should be set to 'Articles & Databases.')
6. Click on a title and then navigate to the user_summon_usage table in the db and note that it has logged your usage. 
7. Navigate back to 'Your Cookie Preferences' and turn off the 'External Search Services' and save. 
8. Check the db again, update the page and notice that the user_summon_usage table has not changed. 
9. In your terminal, run the command adb shell. 
10. Navigate to /usr/local/aspen-discovery/code/cron
11. Run the following command: java -jar cron.jar test.localhostaspen
12. Navigate back to the db and refresh the page/.
13. Notice that they entry in user_summon_usage has now been deleted.
